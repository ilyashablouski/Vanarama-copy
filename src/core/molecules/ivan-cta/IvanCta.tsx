import cx from 'classnames';
import React from 'react';

import Text from 'core/atoms/text';
import Heading from 'core/atoms/heading';
import ImageV2 from 'core/atoms/image/ImageV2';
import Card from 'core/molecules/cards';
import { IBaseProps } from 'core/interfaces/base';

interface IProps extends IBaseProps {
  title: string;
  body: string;
  imageSrc: string;
  imageWidth?: string | number;
  imageHeight?: string | number;
  isCompact?: boolean;
}

const IvanCta: React.FC<IProps> = ({
  isCompact,
  className,
  title,
  body,
  imageSrc,
  imageHeight,
  imageWidth,
  children,
}) => {
  if (isCompact) {
    return (
      <div className={cx(className, 'ivan-cta--compact')}>
        <Text color="white" size="xsmall">
          {body}
        </Text>
        <ImageV2
          width={imageWidth}
          height={imageHeight}
          src={imageSrc}
          alt={title}
          plain
        />
      </div>
    );
  }
  return (
    <Card inline className={cx(className, 'ivan-cta')}>
      <ImageV2
        className="card-image"
        width={imageWidth}
        height={imageHeight}
        src={imageSrc}
        alt={title}
        plain
      />
      <div className="title">
        <Heading size="lead" color="black">
          {title}
        </Heading>
      </div>
      <Text color="dark" size="regular">
        {body}
      </Text>
      {children}
    </Card>
  );
};

export default IvanCta;
