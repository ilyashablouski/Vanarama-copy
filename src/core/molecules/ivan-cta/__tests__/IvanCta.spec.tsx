import React from 'react';
import { shallow } from 'enzyme';
import IvanCta from '../IvanCta';

describe('<IvanCta />', () => {
  it('should default render correctly', () => {
    const wrapper = shallow(
      <IvanCta
        body="Our helpful chatbot iVan is available 24/7 to answer your van financing questions"
        title="Meet iVan"
        imageSrc="https://images.ctfassets.net/3xid768u5joa/31iQbyGS1DYs7b9viQHtXh/86ae2d62793162409d230b2042d9e00a/header-ivan.svg"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should compact render correctly', () => {
    const wrapper = shallow(
      <IvanCta
        isCompact
        body="We're open again at 8:30am. Why not chat to iVan?"
        title=""
        imageSrc="https://www.vanarama.com/assets/images-optimised/header-ivan.svg"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
