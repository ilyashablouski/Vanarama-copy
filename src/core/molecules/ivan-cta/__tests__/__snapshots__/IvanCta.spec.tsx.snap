// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IvanCta /> should compact render correctly 1`] = `
<div
  className="ivan-cta--compact"
>
  <Memo(Text)
    color="white"
    size="xsmall"
  >
    We're open again at 8:30am. Why not chat to iVan?
  </Memo(Text)>
  <Memo(ImageV2)
    alt=""
    plain={true}
    quality={60}
    src="https://www.vanarama.com/assets/images-optimised/header-ivan.svg"
  />
</div>
`;

exports[`<IvanCta /> should default render correctly 1`] = `
<Memo()
  className="ivan-cta"
  inline={true}
>
  <Memo(ImageV2)
    alt="Meet iVan"
    className="card-image"
    plain={true}
    quality={60}
    src="https://images.ctfassets.net/3xid768u5joa/31iQbyGS1DYs7b9viQHtXh/86ae2d62793162409d230b2042d9e00a/header-ivan.svg"
  />
  <div
    className="title"
  >
    <Heading
      color="black"
      size="lead"
    >
      Meet iVan
    </Heading>
  </div>
  <Memo(Text)
    color="dark"
    size="regular"
  >
    Our helpful chatbot iVan is available 24/7 to answer your van financing questions
  </Memo(Text)>
</Memo()>
`;
